docs/proposals/failed-machine-preservation.md

```diff
@@ -0,0 +1,115 @@
+   * For a machine thus annotated, MCM will move it to `Terminating` phase even if `failedMachinePreserveTimeout` has not expired.
+5. If an un-annotated machine moves to `Failed` phase, and the `failedMachinePreserveMax` has not been reached, MCM will auto-preserve this machine.
+6. Machines of a MachineDeployment in `PreserveFailed` stage will also be counted towards the replica count and the enforcement of maximum machines allowed for the MachineDeployment.
+7. At any point in time `machines requested for preservation + machines in PreservedFailed <= failedMachinePreserveMax`. If  `machines requested for preservation + machines in PreservedFailed` is at or exceeds `failedMachinePreserveMax` on annotating a machine, the annotation will be deleted by MCM. 
+
+
+## State Machine
+
+The behaviour described above can be summarised using the state machine below:
+```mermaid
+stateDiagram
+  direction TBP
+  state "PreserveFailed 
+  (node drained)" as PreserveFailed
+  state "Requested 
+  (node & machine annotated)" 
+   as Requested
+  [*] --> Running
+  Running --> Requested:annotated with value=true && max not breached
+  Running --> Running:annotated, but max breached
+  Requested --> PreserveFailed:on failure
+  Running --> PreserveFailed:on failure && max not breached
+  PreserveFailed --> Terminating:after timeout
+  PreserveFailed --> Terminating:annotated with value=false
+  Running --> Failed : on failure && max breached
+  PreserveFailed --> Running : VM recovers
+  Failed --> Terminating
+  Terminating --> [*]
+
+```
+
+In the above state machine, the phase `Running` also includes machines that are in the process of creation for which no errors have been encountered yet.
+
+## Use Cases:
+
+### Use Case 1: Proactive Preservation Request
+**Scenario:** Operator suspects a machine might fail and wants to ensure preservation for analysis.
+#### Steps:
+1. Operator annotates node with `node.machine.sapcloud.io/preserve-when-failed=true`, provided `failedMachinePreserveMax` is not violated
+2. Machine fails later
+3. MCM preserves the machine
+4. Operator analyzes the failed VM
+
+### Use Case 2: Automatic Preservation
+**Scenario:** Machine fails unexpectedly, no prior annotation.
+#### Steps:
+1. Machine transitions to `Failed` phase
+2. If `failedMachinePreserveMax` is not breached, machine moved to `PreserveFailed` phase by MCM
+3. After `failedMachinePreserveTimeout`, machine is terminated by MCM
+
+### Use Case 3: Capacity Management
+**Scenario:** Multiple machines fail when preservation capacity is full.
+#### Steps:
+1. Machines M1, M2 already preserved (failedMachinePreserveMax = 2)
+2. Operator wishes to preserve M3 in case of failure. He increases `failedMachinePreserveMax` to 3, and annotates M3 with `node.machine.sapcloud.io/preserve-when-failed=true`.
+3. If M3 fails, machine moved to `PreserveFailed` phase by MCM.
```

Comment:
FYI changing values directly in the shoot YAML is not always a possibility for operators (E.g. no permissions to edit the shoot YAML (but have admin permission in the cluster). That being said, this is also a valid use case

Reply:
If that is the case then, we can simplify this functionality and completely remove `preserveMachineMax` or what its currently called `failedMachinePreserveMax` (proposal has been iterated several times now but not yet reflected here).
The user can just use annotations on `Node/Machine` to indicate machines that they would like to preserve.
But do remember that it is now the responsibility of whosoever adds the annotations to keep in mind that annotated machines are counted as part of MCD replicas. This means if they do not pay attention and mark multiple machines to be preserved then availability of machines in a MCD can potentially get impacted. The onus of that will solely lie on the operator who made this change.